### PR TITLE
feat: integrate condition on generate

### PR DESCRIPTION
**Pull request by katcipis**
This PR introduces the `condition` functionality specified [here](https://github.com/mineiros-io/terramate/blob/main/docs/codegen/generate-file.md) and [here](https://github.com/mineiros-io/terramate/blob/main/docs/codegen/generate-hcl.md).

### CODE REVIEW

File: generate/generate.go [hunk 1]
```diff
@@ -289,102 +291,54 @@ func (g generatedFiles) add(filename string, genfile generatedFile) error {
...
 
-	logger.Trace().Msg("Checking for outdated generated_file code on stack.")
-
-	stackFiles, err := genfile.Load(root, stackMeta, globals)
-	if err != nil {
-		return err
-	}
-
-	logger.Trace().Msg("Loaded generated_file code, checking")
+	logger.Trace().Msg("Checking for outdated generated_hcl code on stack.")
```
Comment (i4ki) [inline generate/generate.go:304]:
```suggestion
	logger.Trace().Msg("Checking for outdated generated code on stack.")
```

File: generate/generate.go [hunk 2]
```diff
@@ -289,102 +291,54 @@ func (g generatedFiles) add(filename string, genfile generatedFile) error {
...
 
-	for filename, genFile := range stackFiles.GeneratedFiles() {
+	for filename, genHCL := range genfiles {
```
Comment (i4ki) [inline generate/generate.go:306]: why `genHCL`? isn't this reused now for generate_file as well?

File: generate/generate.go [hunk 3]
```diff
@@ -289,102 +291,54 @@ func (g generatedFiles) add(filename string, genfile generatedFile) error {
...
 		logger := logger.With().
 			Str("blockName", filename).
 			Str("targetpath", targetpath).
 			Logger()
 
-		logger.Trace().Msg("Checking if file content is updated.")
+		logger.Trace().Msg("Checking if code is updated.")
 
-		fileContents, found, err := readFile(targetpath)
+		currentHCLcode, codeFound, err := readFile(targetpath)
```
Comment (i4ki) [inline generate/generate.go:315]: same here :thinking:

**Review by i4ki — COMMENTED** [review]
partial review

will checkout the branch and play a bit tomorrow.

File: generate/generate.go (hunk 2 above)
Comment (katcipis) [reply generate/generate.go:306]: ah yeah, forgot to update the name, thanks for the catch :100:

File: generate/generate.go (hunk 3 above)
Comment (katcipis) [reply generate/generate.go:315]: yeah sorry, I refactored it to be general but then messed up fixing the names, thanks for the catch :heart:

**Review by i4ki — APPROVED** [review]
lgtm

**Pull request merged**